### ISSUE TITLE
Missing parts.

### ISSUE THREAD

**Issue by hereket** [matteo-39/vosk-build-model#2]
I tried to follow with your steps and a few issues. It looks like you missed some steps. 
TLDR: The main issue is that when doing everything step by step (fixing missing steps) at the end, when I need to train with "local/chain/tuning/run_tdnn_1j.sh" it says that final.mdl is missing and cannot proceed. 

1. First of all in your description you are missing spk2utt file creation and because of this "utils/validate_data_dir.sh data/train" reports and error about missing file. So you are missing a step of creating utt2spk and later generating spk2utt file.
2. You do not create feats.scp and other files that are generated with "./steps/makemfcc ..."
3. Your forgot to create cmvn.scp with "steps/compute_cmvn_stats.sh ..."

**Comment by matteo-convertino** [comment]
Regarding missing files, it is my mistake because when I say, in the [Data Creation](https://github.com/matteo-39/vosk-build-model#data-creation), "By following this official kaldi guide" I thought the reader did the steps in that guide on his own (like creating all those necessary files). Thanks for pointing it out to me because now I know that I have to be more specific.

As for the main issue, can you post here the logs? Because I remember that the final.mdl file is created during training, so it's strange that it says this file is missing.

if it helps you, I only had one error with the final.mdl file. The problem was that the file **was present** and therefore had to be removed to continue the training (basically the opposite of what you told me).

**Comment by hereket** [comment]
The final.mdl is there now as I supplied all the required files. It was just when I followed this step by step errors got lost in a long list of messages. The I read all the messages one by one and fixed all errors and missing files.

But now I get this:
```
...
...
...
LOG (ivector-extractor-est[5.5.0~1539-72c1]:UpdateProjections():ivector-extractor.cc:1431) Overall objective function improvement for M (mean projections) was 0 per frame over 10166 frames.
ASSERTION_FAILED (ivector-extractor-est[5.5.0~1539-72c1]:UpdateVariances():ivector-extractor.cc:1486) Assertion failed: (var_floor_count > 0.0)

[ Stack-Trace: ]
/opt/kaldi/src/lib/libkaldi-base.so(kaldi::MessageLogger::LogMessage() const+0xa39) [0x7f84141ced3d]
/opt/kaldi/src/lib/libkaldi-base.so(kaldi::KaldiAssertFailure_(char const*, char const*, int, char const*)+0x59) [0x7f84141cf991]
/opt/kaldi/src/lib/libkaldi-ivector.so(kaldi::IvectorExtractorStats::UpdateVariances(kaldi::IvectorExtractorEstimationOptions const&, kaldi::IvectorExtractor*) const+0x449) [0x7f8414433d6d]
/opt/kaldi/src/lib/libkaldi-ivector.so(kaldi::IvectorExtractorStats::Update(kaldi::IvectorExtractorEstimationOptions const&, kaldi::IvectorExtractor*) const+0x1eb) [0x7f84144397b3]
ivector-extractor-est(main+0x76d) [0x403b33]
/usr/lib/libc.so.6(__libc_start_main+0xd5) [0x7f840cb5ab25]
ivector-extractor-est(_start+0x2e) [0x4032ee]
```

Could this be due to low amount of data? I am using only 85 utterances with only digits.

**Comment by matteo-convertino** [comment]
Yes, for training 85 utterances are very few. For example my lowest number of sentences used was about 10000. Also in the dataset that downloads the recipe by default (http://www.openslr.org/resources/31/, http://www.openslr.org/resources/11/) about 200000 sentences are used.
I don't know how much it is and if there is a minimum. 

Try to find a dataset in the language you want and then add your utterances to it (you can also check the link in the guide).
But unfortunately you don't always find a dataset prepared for training with kaldi and you have to organise the files manually.
